baseline-shift initial value should be `0px` rather than `baseline`

Ahmad-S792 opened the pull request:
#### 6191387a5dda9101bb3e5a67429ec31583420a6b
<pre>
baseline-shift initial value should be `0px` rather than `baseline`

<a href="https://bugs.webkit.org/show_bug.cgi?id=269878">https://bugs.webkit.org/show_bug.cgi?id=269878</a>
<a href="https://rdar.apple.com/problem/123792212">rdar://problem/123792212</a>

Reviewed by NOBODY (OOPS!).

This patch aligns WebKit with Blink / Chromium and Web Specification [1]:

[1] <a href="https://drafts.csswg.org/css-inline-3/#property-index">https://drafts.csswg.org/css-inline-3/#property-index</a>

As per web specification, the initial value for &apos;baseline-shift&apos; is &apos;0&apos; of `&lt;length-percentage&gt;`.

* Source/WebCore/css/CSSProperties.json:
Set the initial value of `baseline-shift` to `0px`.
* LayoutTests/imported/w3c/web-platform-tests/css/css-inline/inheritance-expected.txt: Rebaselined
* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt: Rebaselined
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt: Ditto
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-presentation-attribute-expected.txt: Ditto (failure matching with other browsers)
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt:
* LayoutTests/svg/css/scientific-numbers.html: Updated
* LayoutTests/svg/css/scientific-numbers-expected.txt: Rebaselined
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6191387a5dda9101bb3e5a67429ec31583420a6b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/170240 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/44163 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/168/builds/37580 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/179613 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/127952 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/7d1c5f11-c722-4218-92df-52e245f76e78) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/172106 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/45407 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/43795 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/132725 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/93550 "1 flakes 14 failures") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/a8e8d871-5900-46cf-8137-072535055d06) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/173199 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/35902 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/153144 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/113226 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/9e3e4f99-fedf-4da3-8a91-4b42e965e839/626df86f-f524-4bae-abf8-28c7c8f08404) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/34491 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/32851 "Passed tests") | [✅ 🛠 gtk3-libwebrtc](https://ews-build.webkit.org/#/builders/173/builds/28298 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/144974 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/169/builds/32542 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/182199 "Built successfully") | | 
| | [✅ 🛠 ios-safer-cpp](https://ews-build.webkit.org/#/builders/174/builds/34988 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/37023 "Found 1 new test failure: http/tests/site-isolation/https-load.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/141166 "Passed tests") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/43820 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/152614 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/140990 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/44509 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/170/builds/29526 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/108920 "Built successfully") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/36262 "Passed tests") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/116390 "Built successfully") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/43019 "Built successfully") | [✅ 🧪 mac-site-isolation](https://ews-build.webkit.org/#/builders/185/builds/7626 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/42411 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/42665 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/42554 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

Ahmad-S792 committed:
baseline-shift initial value should be `0px` rather than `baseline`

https://bugs.webkit.org/show_bug.cgi?id=269878
rdar://problem/123792212

Reviewed by NOBODY (OOPS!).

This patch aligns WebKit with Blink / Chromium and Web Specification [1]:

[1] https://drafts.csswg.org/css-inline-3/#property-index

As per web specification, the initial value for 'baseline-shift' is '0' of `<length-percentage>`.

* Source/WebCore/rendering/style/SVGRenderStyle.h:
(initialBaselineShift):
(initialBaselineShiftValue):
* LayoutTests/imported/w3c/web-platform-tests/css/css-inline/inheritance-expected.txt: Rebaselined
* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt: Rebaselined
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt: Ditto
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-presentation-attribute-expected.txt: Ditto (failure matching with other browsers)
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt:
* LayoutTests/svg/css/scientific-numbers.html: Updated
* LayoutTests/svg/css/scientific-numbers-expected.txt: Rebaselined

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0c10fe778515384507f52ec520e3099667c426a0. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/24908

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/53e7bfd63e66a30f1264b7abab30fdca74d65952. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/24908

karlcow commented:
@graouts @shallawa Anything which is holding this PR?

Ahmad-S792 commented:
> @graouts @shallawa Anything which is holding this PR?

I think we started failing more `scientific-notation` test and return wrong. So I think there is still something wrong, my bad, I should've changed it to 'draft'.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9787adc4424d07f83a8f35320ce497666a272e6b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/24908

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e1d25c189bbb177705b476fcd8d90905269623b2. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/24908

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/544bc91e0e628f1b1fb031ae3917c6a3e1f673b4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/24908

smfr commented on LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-presentation-attribute-expected.txt:
```diff
@@ -4,7 +4,7 @@ PASS Testing 'stroke-width' on '#box2'.
 PASS Testing 'clip' on '#test4'.
 PASS Testing 'alignment-baseline'.
-PASS Testing 'baseline-shift'.
+FAIL Testing 'baseline-shift'. assert_equals: Default value. expected "baseline" but got "0px"
```
Why this new failure?

Ahmad-S792 replied:
@smfr - this test is wrong on WPT upstream, it fails across the board. It should be expecting '0px' as per spec but it expect 'baseline' (old value).

Ahmad-S792 replied:
WPT test case - https://wpt.fyi/results/css/css-variables/variable-presentation-attribute.html?label=experimental&label=master&aligned

Ahmad-S792 committed:
baseline-shift initial value should be `0px` rather than `baseline`

https://bugs.webkit.org/show_bug.cgi?id=269878
rdar://problem/123792212

Reviewed by NOBODY (OOPS!).

This patch aligns WebKit with Blink / Chromium and Web Specification [1]:

[1] https://drafts.csswg.org/css-inline-3/#property-index

As per web specification, the initial value for 'baseline-shift' is '0' of `<length-percentage>`.

* Source/WebCore/css/CSSProperties.json:
Set the initial value of `baseline-shift` to `0px`.
* LayoutTests/imported/w3c/web-platform-tests/css/css-inline/inheritance-expected.txt: Rebaselined
* LayoutTests/fast/css/getComputedStyle/computed-style-expected.txt: Rebaselined
* LayoutTests/fast/css/getComputedStyle/computed-style-without-renderer-expected.txt: Ditto
* LayoutTests/imported/w3c/web-platform-tests/css/css-variables/variable-presentation-attribute-expected.txt: Ditto (failure matching with other browsers)
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt:
* LayoutTests/svg/css/scientific-numbers.html: Updated
* LayoutTests/svg/css/scientific-numbers-expected.txt: Rebaselined
